app/models/plugins/has_form.rb

```diff
@@ -21,6 +21,15 @@ def outstanding_fields(form_values)
     ).errors.keys
   end
 
+  def update_form(new_form)
+    if form
```

Comment:
wouldn't it make sense to check that the new_form is different than the old form?